infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/LogReplicationSourceManager.java

```diff
@@ -56,39 +49,19 @@ public class LogReplicationSourceManager {
     @VisibleForTesting
     private ObservableAckMsg ackMessages = new ObservableAckMsg();
 
-    /**
-     * Constructor
-     *
-     * @param params Log Replication parameters
-     * @param metadataManager Replication Metadata Manager
-     */
-    public LogReplicationSourceManager(LogReplicationRuntimeParameters params, IClientServerRouter router,
+    private boolean isShutdown = false;
+
+
+    public LogReplicationSourceManager(IClientServerRouter router,
                                        LogReplicationMetadataManager metadataManager,
                                        LogReplicationSession session, LogReplicationContext replicationContext) {
-        this(params, metadataManager, new CorfuDataSender(router, session), session, replicationContext);
+        this(metadataManager, new CorfuDataSender(router, session), session, replicationContext);
     }
 
     @VisibleForTesting
-    public LogReplicationSourceManager(LogReplicationRuntimeParameters params,
-                                       LogReplicationMetadataManager metadataManager, DataSender dataSender,
+    public LogReplicationSourceManager(LogReplicationMetadataManager metadataManager, DataSender dataSender,
                                        LogReplicationSession session, LogReplicationContext replicationContext) {
 
-        // This runtime is used exclusively for the snapshot and log entry reader which do not require a cache
-        // as these are one time operations.
-        this.runtime = CorfuRuntime.fromParameters(CorfuRuntimeParameters.builder()
-                .trustStore(params.getTrustStore())
-                .tsPasswordFile(params.getTsPasswordFile())
-                .keyStore(params.getKeyStore())
-                .ksPasswordFile(params.getKsPasswordFile())
-                .systemDownHandler(params.getSystemDownHandler())
-                .tlsEnabled(params.isTlsEnabled())
-                .cacheDisabled(true)
```

Comment:
similar point here.  The cache is not disabled in the runtime passed from InterclusterReplicationServer

Reply:
Same...we need to consider this while tuning the cache. And also consider having a runtime per model, so cache is disabled for models where FSMs will not be sharing the data to send, and enabled for models where multiple FSMs share data to be replicated